### PR TITLE
Added :meth:`~.CoordinateSystem.get_line_graph` for plotting a line graph

### PR DESCRIPTION
**Pull request by abhi3019**
## Changelog / Overview

Added :meth:`~.CoordinateSystem.get_line_graph` that returns a line graph from lists of points along x, y and z (optional) axes.

## Motivation

Allows plotting line graphs easily from lists of values along axes.

## Example
- code
```py
class LineGraphScene(Scene):
    def construct(self):
        axes = NumberPlane(y_min=-8, y_max=8, y_axis_config={"width":7})
        axes.set_opacity(0.25)
        
        x_values = np.linspace(-6, 6, 20)
        y_values = [np.random.random()*8*(-1)**i for i in range(20)]
        lg = axes.get_line_graph(
            x_values, 
            y_values, 
            line_color=GOLD_D, 
            vertex_dot_style=dict(fill_opacity=0, stroke_width=4),
            stroke_width=2
        )

        self.add(axes)
        self.wait()
        self.play(Create(lg["vertex_dots"], lag_ratio=1, run_time=2))
        self.wait(0.5)
        self.play(Create(lg["line_graph"], run_time=4, rate_func=linear))
        self.wait(0.5)
```
- result

https://user-images.githubusercontent.com/73361366/115595485-9a565080-a2f4-11eb-8085-d50fddd05c52.mp4

       

## Testing Status

- `pytest` - local pass (Windows 10)
- `pytest --doctest-modules manim` - doc tests fail (on master as well), the first fail happens here:
```py
================================================== FAILURES ================================================== 
________________________ [doctest] manim.mobject.svg.opengl_tex_mobject.OpenGLMathTex ________________________ 
366         class Formula(Scene):
367             def construct(self):
368                 t = MathTex(r"\int_a^b f'(x) dx = f(b)- f(a)")
369                 self.add(t)
370
371     Tests
372     -----
373     Check that creating a :class:`~.MathTex` works::
374
375         >>> MathTex('a^2 + b^2 = c^2')
UNEXPECTED EXCEPTION: NameError("name 'MathTex' is not defined")
``` 
## Further Comments

In future, I'd like to see a `LineGraph` (a `VDict` preferably) with more options such as dashed lines, legends,  and so on, but for now this is good enough for `CoordinateSystem`s imo.

## Checklist
- [x] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)
- [x] I have written a descriptive PR title (see top of PR template for examples)
- [x] I have written a changelog entry for the PR or deem it unnecessary
- [ ] My new functions/classes either have a docstring or are private
- [ ] My new functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] My new documentation builds, looks correctly formatted, and adds no additional build warnings

## Reviewer Checklist
- [x] The PR title is descriptive enough
- [x] The PR is labeled correctly
- [x] The changelog entry is completed if necessary
- [x] Newly added functions/classes either have a docstring or are private
- [ ] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [x] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings

### CODE REVIEW

**Review by behackl — CHANGES_REQUESTED** [review]
The implementation looks good to me, but I'd like to see a bit of documentation for the method -- you don't necessarily have to include an (animated) example (although it would help), but there should be a basic docstring containing a one-line summary as well as a "Parameters"-section briefly explaining all of the parameters that can be passed.

(And for extra internet points, type hints for the arguments and return type would also be nice.)

**Comment by abhi3019** [comment]
Hope the typings are correct and the docstrings make sense. I've also moved this method into `Axes` because `CoordinateSystem` sounds more general and it doesn't really make sense to have this method for say a `PolarCoordinateSystem`

File: manim/mobject/coordinate_systems.py [hunk 1]
```diff
@@ -212,6 +213,85 @@ def add_coordinates(self, x_vals=None, y_vals=None):
...
+        line_color : :class:`Colors`
+            Color for the line graph.
+        add_vertex_dots : :class:`bool`
+            Whether or not to add :class:`~.Dot` at each vertex.
+        vertex_dot_radius : :class:`float`
+            Radius for the :class:`~.Dot` at each vertex.
+        vertex_dot_style : :class:`dict`
+            Style arguments to be passed into :class:`~.Dot` at each vertex.
+        kwargs : Any
+            Additional arguments to be passed into :class:`~.VMobject`.
```
Comment (behackl) [inline manim/mobject/coordinate_systems.py:248]: Typings don't have to be added in both docstring and method header.

```suggestion
        x_values
            Iterable of values along the x-axis.
        y_values
            Iterable of values along the y-axis.
        z_values
            Iterable of values (zeros if z_values is None) along the z-axis.
        line_color
            Color for the line graph.
        add_vertex_dots
            Whether or not to add :class:`~.Dot` at each vertex.
        vertex_dot_radius
            Radius for the :class:`~.Dot` at each vertex.
        vertex_dot_style
            Style arguments to be passed into :class:`~.Dot` at each vertex.
        kwargs
            Additional arguments to be passed into :class:`~.VMobject`.
```

**Review by behackl — APPROVED** [review]
Looks great, thanks! I've only made minor suggestions to the documentation, I'll commit these myself, check that the documentation renders successfully and correctly -- and if so, merge the PR.

File: manim/mobject/coordinate_systems.py [hunk 2]
```diff
@@ -212,6 +213,87 @@ def add_coordinates(self, x_vals=None, y_vals=None):
...
+        line_color: Colors = YELLOW,
+        add_vertex_dots: bool = True,
+        vertex_dot_radius: float = DEFAULT_DOT_RADIUS,
+        vertex_dot_style: Optional[dict] = None,
+        **kwargs
+    ) -> VDict:
+        """Draws a line graph.
+
+        The graph connects the vertices formed from zipping
+        ``x_values``, ``y_values`` and ``z_values``. Also adds :class:`~.Dot`s at the
```
Comment (behackl) [inline manim/mobject/coordinate_systems.py:230]:
```suggestion
        ``x_values``, ``y_values`` and ``z_values``. Also adds :class:`~.Dot`_s at the
```

**Comment by abhi3019** [comment]
Thanks for fixing the issues!

**Pull request merged**